### PR TITLE
Allow switching VM power state during restore 

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

This Pull Request enhances the behavior of VirtualMachine restores by introducing a new label:

    "velero.kubevirt.io/restore-run-strategy"

This label can be added to the restore resource to explicitly control the power state of the VirtualMachine during the restore process. The VirtualMachine's state is backed up as it exists at the time of backup, but these labels allow you to override the default behavior and define the desired RunStrategy at restore time.

**Which issue(s) this PR fixes**:
Fixes # https://github.com/kubevirt/kubevirt-velero-plugin/issues/294

**Special notes for your reviewer**:

There's some overlapping code with https://github.com/kubevirt/kubevirt-velero-plugin/pull/304 but conflicts will be easy to solve.

**Release note**:

```release-note
Allow switching VM run strategy during restore 
```

### CODE REVIEW

**Comment by mhenriks** [comment]
Would it be better to have a single label that maps to the VM `runStrategy`?

**Comment by alromeros** [comment]
> Would it be better to have a single label that maps to the VM `runStrategy`?

That's so much better, thanks for the suggestion. Will address it on Monday.

File: pkg/plugin/vm_restore_item_action.go [hunk 1]
```diff
@@ -63,6 +65,11 @@ func (p *VMRestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (
 		return nil, errors.WithStack(err)
 	}
 
+	if runStrategy, ok := util.GetRestoreRunStrategy(input.Restore); ok {
+		p.log.Infof("Setting virtual machine run strategy to %s", runStrategy)
+		vm.Spec.RunStrategy = ptr.To(runStrategy)
```
Comment (mhenriks) [inline pkg/plugin/vm_restore_item_action.go:70]: should make sure to unset `spec.running` when using runStrategy pretty sure webhook will deny otherwise

Comment (alromeros) [reply pkg/plugin/vm_restore_item_action.go:70]: ACK

**Review by awels — APPROVED** [review]

**Pull request merged**